### ISSUE TITLE
Stop Loss not triggered

### ISSUE THREAD

**Issue by 0jessica0** [freqtrade/freqtrade#10710]
## Describe your environment
Operating system: ____windows（no docker）
Python Version: _____3.11.9 (python -V)
CCXT version: _____4.3.98 (pip freeze | grep ccxt)
Freqtrade Version: ____2024.9 (freqtrade -V or docker compose run --rm freqtrade -V for Freqtrade running in docker)
  
## Your question

Question 1:
how to calculate the stop_loss_abs?
Here is the setting in the strategy:
```
stoploss = -0.5
use_custom_stoploss = True
# Trailing stoploss
trailing_stop = True
trailing_stop_positive = 0.02
trailing_stop_positive_offset = 0.03  # Disabled / not configured
trailing_only_offset_is_reached = True
```
```
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                            current_rate: float, current_profit: float, **kwargs) -> float:

        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        stoploss = 999999

        # Convert absolute price to percentage relative to current_rate
        if stoploss < current_rate:
            return (stoploss / current_rate) - 1

        # return maximum stoploss value, keeping current stoploss price unchanged
        return 1
```

Here is an example in backtest:
```
{'pair': 'RAY/USDT',
 'stake_amount': 198.78905,
 'max_stake_amount': 198.78905,
 'amount': 1435.3,
 'open_date': '2023-01-01 18:45:00+00:00',
 'close_date': '2023-01-01 19:00:00+00:00',
 'open_rate': 0.1385,
 'close_rate': 0.1387,
 'fee_open': 0.001,
 'fee_close': 0.001,
 'trade_duration': 15,
 'profit_ratio': -0.0005568438781435171,
 'profit_abs': -0.11080516,
 'exit_reason': 'trailing_stop_loss',
 'initial_stop_loss_abs': 0.07,
 'initial_stop_loss_ratio': -0.5,
 'stop_loss_abs': 0.139,
 'stop_loss_ratio': -0.004428880420228376,
 'min_rate': 0.1385,
 'max_rate': 0.1387,
 'is_open': False,
 'enter_tag': '',
 'leverage': 1.0,
 'is_short': False,
 'open_timestamp': 1672598700000,
 'close_timestamp': 1672599600000,
 'orders': [{'amount': 1435.3,
   'safe_price': 0.1385,
   'ft_order_side': 'buy',
   'order_filled_timestamp': 1672598700000,
   'ft_is_entry': True,
   'ft_order_tag': ''},
  {'amount': 1435.3,
   'safe_price': 0.1387,
   'ft_order_side': 'sell',
   'order_filled_timestamp': 1672599600000,
   'ft_is_entry': False,
   'ft_order_tag': 'trailing_stop_loss'}]}
```

Can you tell me how to calculate the price of stop_loss_abs?

Question 2:
This question is quite urgent and critical:
Why I didn't change anything, but I can get exit reason of 'trailing_stop_loss' in backtest, but no exit in live trade?
I didn't change anything between backtest and live trading.
The stop_loss is still -50%, unlike in the backtest within 3%

Question 3:
how to trigger force_exit?

Thanks

**Comment by xmatthias** [comment]
well as the [documentation explains](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) (green box at the bottom of this section) - we strongly advise not to use `custom_stoploss()` with `trailing_stop` together.
The two can issue conflicting settings - with the "tighter" one winning.
Your stoploss hence was not 3% - but something way tighter due to custom stop.

leaving that aside - your resulting stoploss is **way** too tight (independent of all other points, percentage is not subject to rounding).

After trailing (most likely custom stoploss, really) - your stop results in ` 'stop_loss_ratio': -0.004428880420228376,` - which is 0.04% below current price.
Even using `price=other` in the configuration will mean your stoploss is now ABOVE the current price - in theory causing an immediate trigger - in practice causing the stoploss placement to fail (for stoploss on exchange) - resulting in the trade to be force-exited.

This tight a stoploss is also gaming backtesting - as it'll cause it to adjust to high and immediately trigger (at high - stop).
While possible in theory - this is a great way to fake great profits - which won't replicate in dry-run in our experience.

---

The price itself (`'stop_loss_abs': 0.139,`) is caused by precision rounding. RAY/USDT has a precision of 0.001 (assuming binance) - so price can move in ticks of 0.001 - hence prices need rounding to that value.
These limits do change over time (as price changes) - but there's no way to access "historic" limits.
the problematic in this is explained in [trading limits in backtesting](https://www.freqtrade.io/en/stable/backtesting/#trading-limits-in-backtesting).

---

> how to trigger force_exit?

in backtesting - you don't. It's automatic if a trade is still open at the end of the backtest period (or when the data stops).
in live - use telegram or freqUI.

**Comment by stash86** [comment]
For q2, the issue is simple, your "if" check never satisfied. In backtest, trades are assumed to be opened at candle open, which means your "if" condition will be correct. For dry/live, you will never get trade open at candle open.

**Comment by stash86** [comment]
Read this on how to access signal candle and get indicators' value to be used for custom exit/stoploss

https://botacademy.ddns.net/2024/04/18/fsd-better-way-of-using-atr-for-exits/

**Comment by 0jessica0** [comment]
> Read this on how to access signal candle and get indicators' value to be used for custom exit/stoploss
> 
> https://botacademy.ddns.net/2024/04/18/fsd-better-way-of-using-atr-for-exits/

wow, it is very detailed.
Are you suggesting that the custom_stoploss is actually an exit?
Just move it to exit and remove the custom_stoploss will work?
